src/main/java/com/sequenceiq/cloudbreak/service/template/SimpleTemplateService.java

```diff
@@ -111,20 +100,27 @@ public Template getPublicTemplate(String name, CbUser user) {
     }
 
     @Override
-    public void delete(String templateName, CbUser cbUser) {
-        Template template = templateRepository.findByNameInAccount(templateName, cbUser.getAccount(), cbUser.getUserId());
+    public void delete(String templateName, CbUser user) {
+        Template template = templateRepository.findByNameInAccount(templateName, user.getAccount(), user.getUserId());
+        delete(template, user);
+    }
+
+    private void delete(Template template, CbUser user) {
+        MDCBuilder.buildMdcContext(template);
         if (template == null) {
-            throw new NotFoundException(String.format(TEMPLATE_NOT_FOUND_MSG, templateName));
+            throw new NotFoundException(String.format(TEMPLATE_NOT_FOUND_MSG, template.getId()));
```

Comment:
Users can identify the templates by name, so the error message should contain the name of the template and the template id or just the template name.